### ISSUE TITLE
Larger font sizes cause layout issues for AppBar

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#36265]
Internal: b/140826056

Using standard material components, increasing the font size causes layout artifacts. See this example:

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        theme: ThemeData(primaryColor: Colors.blue),
        home: Scaffold(
          appBar: AppBar(
            title: Text('Menu Test'),
            actions: [
            PopupMenuButton<String>(
              itemBuilder: (context) {
                final result =List<PopupMenuEntry<String>>();
                for(int i = 0; i < 5; ++i) {
                  result.add(PopupMenuItem<String>(value: '$i', child: Text('Menu $i')));
                }
                return result;
              },
            )
            ]
          ),
        ));
  }
}
```

This yields to:

![image](https://user-images.githubusercontent.com/6687929/61305987-a5d02b80-a7a0-11e9-98f8-0099e943a100.png)

Similar bugs have been filed for specific components such as: https://github.com/flutter/flutter/issues/5939.

What is the recommended solution to this? Material guidelines state that the UI layout should scale nicely with font size: https://material.io/design/usability/accessibility.html#implementing-accessibility.

**Comment by mehmetf** [comment]
In the particular example above, I would expect:

- The app bar to pad the tile.
- The overflow menu to show menu items with padding.
- The gray selection bar to cover the entire text and padding.

**Comment by Hixie** [comment]
In general we do handle this, but some specific components fail to do it right. The most useful thing to do would be to file specific bugs on each case you find. (We added text zoom late in the game and have not done a perfect job of retrofitting it.)

**Comment by Hixie** [comment]
cc @HansMuller

**Comment by HansMuller** [comment]
The test case, as is, running on Android 9 with the text scaling at its highest setting, appears to do the right thing. [Test case source](https://gist.github.com/HansMuller/9b8963a6ee9ec1edda58d08f68745f3a
).

![big_menu](https://user-images.githubusercontent.com/1377460/63971188-c9062000-ca5a-11e9-8b0d-f9e9ee1b2604.png)

One can see the reported problem by setting the text style of the menu items to something big, if the item heights and the menu's padding are not specified. Both of these parameters have constant default values which are too small for jumbo menu items. [Test case source](https://gist.github.com/HansMuller/1630a9f54a1e14d29e09918e0e3348fa).

![big_menu_fail](https://user-images.githubusercontent.com/1377460/63971252-f5ba3780-ca5a-11e9-82f2-8155a7c3e9eb.png)

One can accommodate jumbo menu item sizes by specifying the menu's padding and the popup menu items' heights. [Test case source](https://gist.github.com/HansMuller/a5de8483d68090a46a9182cb23ce5df8).

![big_menu_ok](https://user-images.githubusercontent.com/1377460/63971264-fb178200-ca5a-11e9-8b36-69d62970767e.png)

This shouldn't be necessary, i.e. we can compute reasonable values in this case.

I've opened a new issue for that limitation: https://github.com/flutter/flutter/issues/39508

**Comment by mehmetf** [comment]
Hans, could you please test this also on iOS? I filed the bug thinking this is a platform agnostic issue but as you said, it does not reproduce on Android. I still think it is odd because the tap target does not vertically center the text (you can see it in your sshot) but it is not nearly as bad as on iOS.

**Comment by HansMuller** [comment]
The menu problem does reproduce on Android, however it's possible to work around the issue by specifying popup menu button padding, and item heights (that's the second and third cases above). I will test https://github.com/flutter/flutter/issues/39508 on both platforms.

I haven't looked at the AppBar issue yet.

**Comment by QuattroNinos** [comment]
Internally tracked in b/140826056

**Comment by tvolkert** [comment]
Customer Money (g3) needs this fixed by mid-Q4.

**Comment by HansMuller** [comment]
The menu item issue has been fixed, see https://github.com/flutter/flutter/pull/40179. I updated b/140826056

**Comment by Hixie** [comment]
So what is left?

**Comment by HansMuller** [comment]
@Hixie https://github.com/flutter/flutter/issues/36265#issuecomment-511858371 - also notes that the AppBar's title doesn't fit so (perhaps) we should automatically move it upwards. Still TBD.

**Comment by Hixie** [comment]
Ok, I'll update the issue title then.

**Comment by HansMuller** [comment]
I've created a separate issue for the jumbo appbar title issue: https://github.com/flutter/flutter/issues/42932. I'll close both of them when https://github.com/flutter/flutter/pull/42936 lands.
